workflow/controller/operator.go

```diff
@@ -1309,7 +1309,15 @@ func inferFailedReason(pod *apiv1.Pod) (wfv1.NodePhase, string) {
 			continue
 		}
 
-		msg := fmt.Sprintf("exit code %d: %s; %s; %s", t.ExitCode, t.Reason, t.Message, annotatedMsg)
+		msg := fmt.Sprintf("%s (exit code %d)", t.Reason, t.ExitCode)
+		splitChar := ":"
+		if t.Message != "" {
+			msg = fmt.Sprintf("%s%s %s", msg, splitChar, t.Message)
+			splitChar = ","
+		}
+		if annotatedMsg != "" && annotatedMsg != t.Message {
+			msg = fmt.Sprintf("%s%s %s", msg, splitChar, annotatedMsg)
```

Comment:
I think we should get rid of `annotatedMsg` completely - because `ContainerStatus.Message` is usually the same thing, but not for this PR.